src/inference/include/openvino/runtime/intel_gpu/properties.hpp

```diff
@@ -71,6 +71,15 @@ static constexpr Property<bool> enable_loop_unrolling{"GPU_ENABLE_LOOP_UNROLLING
  */
 static constexpr Property<bool> disable_winograd_convolution{"GPU_DISABLE_WINOGRAD_CONVOLUTION"};
 
+/**
+ * @brief Turning on this key enables full model caching.
+ * Default model caching policy serializes only OpenCL kernels. Full model caching serializes OpenCL kernels and
+ * all types of parameters. If faster loading time is a priority over smaller cache file size,
+ * full model caching should be enabled.
+ * @ingroup ov_runtime_ocl_gpu_prop_cpp_api
+ */
+static constexpr Property<bool> enable_full_model_caching{"GPU_ENABLE_FULL_MODEL_CACHING"};
```

Comment:
1. I think we need a better name. maybe `cache_mode` = `{optimize_size, optimize_speed}`
2. Can it be a common property? 
 
@ilya-lavrenov @isanghao what do you think?

Reply:
Good questions. 

Personally, I do expect this property to be a temporary solution until we can reduce model cache size. But it's common problem between all OpenVINO inference devices. For example, CPU could completely turn off caching if `optimize_size` is specified. 

So, I vote for a common property, which will be deprecated and removed once we solve issues with cache (unless new use-cases for this property are emerged)